src/main/java/gregtech/api/capability/impl/FuelRecipeLogic.java

```diff
@@ -131,24 +155,50 @@ public void update() {
         }
     }
 
+    /**
+     * Whether the multiblock can begin searching for new recipes once a recipe is completed.
+     * This is called only when the multiblock begins searching for new recipes, not during the recipe progression.
+     * This method will also cancel searching for recipes if there are too many maintenance issues.
+     *
+     * @return {@code true} if the multiblock should search for a new recipe
+     */
     protected boolean isReadyForRecipes() {
-        return true;
+        if (metaTileEntity instanceof MultiblockWithDisplayBase) {
+
+            MultiblockWithDisplayBase controller = (MultiblockWithDisplayBase) metaTileEntity;
+
+            // do not run recipes when there are more than 5 maintenance problems
+            if (controller.hasMaintenanceMechanics() && controller.getNumMaintenanceProblems() > 5) {
+                return false;
+            }
```

Comment:
Instead of checking a specific MTE class, add the `hasMaintenanceMechanics()` method to the `IMaintenance` interface, and cast to that interface instead. This leave it more universal.